File: backend/profiles/serializers/friendship_serializer.py
from rest_framework import serializers

from ..models import Friendship
from .user_serializer import UserSerializer


class FriendshipSerializer(serializers.ModelSerializer):
    user = UserSerializer(many=False, read_only=True)
    friend = UserSerializer(many=False, read_only=True)

    class Meta:
        model = Friendship
        fields = ['id', 'user', 'friend']

File: backend/app/serializers/api/v1/friendship_serializer.rb
module Api
  module V1
    class FriendshipSerializer < ApplicationSerializer
      type 'friendships'

      attributes :created_at

      belongs_to :user, serializer: versioned_class(UserSerializer)
      belongs_to :friend, serializer: versioned_class(UserSerializer)

      link(:self) do
        methods = []
        methods << 'DELETE' if Friendships::Destroy.can_run?(friendship: object)

        {
          href: scope.versioned_url_for(:friendship, object),
          meta: {
            methods:
          }
        }
      end
    end
  end
end
